impl/src/main/java/org/glassfish/json/JsonParserImpl.java

```diff
@@ -501,6 +501,10 @@ void skip() {
                     case CURLYOPEN:
                         depth++;
                         break;
+                    case EOF:
+                        throw parsingException(JsonToken.EOF, "[CURLYCLOSE]");
```

Comment:
this is not exactly true per JSON grammar since after  'CURLYOPEN' there can be also 'STRING'. Generally speaking, the error should list all valid options.